modules/account-lib/src/coin/hbar/coinTransferBuilder.ts

```diff
@@ -0,0 +1,130 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import * as Long from 'long';
+import * as proto from '@hashgraph/proto';
+import { InvalidParameterValueError } from '@bitgo/sdk-core';
+import { TransferBuilder, Recipient } from './transferBuilder';
+import { Transaction } from './transaction';
+import { isValidAddress, isValidAmount, stringifyAccountId } from './utils';
+import { BigNumber } from 'bignumber.js';
+
+export class CoinTransferBuilder extends TransferBuilder {
+  // TODO: [BG-51282] Deprecate once wp work for multi recipients
+  private _toAddress: string;
+  private _amount: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.transfers = this.buildTransferData();
+    return await super.buildImplementation();
+  }
+
+  private buildTransferData(): proto.ITransferList {
+    let totalSend = new BigNumber(0);
+    const accountAmounts: proto.AccountAmount[] = [
+      {
+        accountID: this.buildAccountData(this._source.address),
+        amount: Long.fromInt(0),
+      }, // sender
+    ];
+    // add recipients and update send amount
+    this._recipients.map((recipient) => {
+      accountAmounts.push(
+        { accountID: this.buildAccountData(recipient.address), amount: Long.fromString(recipient.amount) }, // recipient
+      );
+      totalSend = totalSend.plus(recipient.amount);
+    });
+    accountAmounts[0].amount = Long.fromString(totalSend.toString()).negate(); // update sender send amount
+
+    return {
+      accountAmounts: accountAmounts,
+    };
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.transfers && transferData.transfers.accountAmounts) {
+      this.initTransfers(transferData.transfers.accountAmounts);
+    }
+  }
+
+  /**
+   * Initialize the transfer specific data, getting the recipient account
+   * represented by the element with a positive amount on the transfer element.
+   * The negative amount represents the source account, so it's ignored.
+   *
+   * @param {proto} transfers - Array of objects which contains accountID and transferred amount
+   */
+  protected initTransfers(transfers: proto.IAccountAmount[]): void {
+    transfers.forEach((transferData) => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        this.send({
+          address: stringifyAccountId(transferData.accountID!),
+          amount: amount.toString(),
+        });
+      }
+    });
+  }
+
+  // region Transfer fields
+  /** TODO: [BG-51282] Deprecate to and amount once wp has been fixed
+      Currently work for one recipient by using exatcly one of to + amount or send function
```

Comment:
In the mean time, I think you should enforce this point. Add a check before `to`, `amount` and `send` methods to fail if recipients are set and vice versa.